### PR TITLE
[soltest] Improve Builtin Initialization.

### PR DESCRIPTION
**Pull request by aarlt**
Based on @cameel comment @ https://github.com/ethereum/solidity/pull/11048/files#r624194465.

### CODE REVIEW

File: test/libsolidity/SemanticTest.cpp [hunk 1]
```diff
@@ -118,29 +117,37 @@ SemanticTest::SemanticTest(
...
-	};
-	soltestAssert(m_builtins.count("storageEmpty") == 0, "");
-	m_builtins["storageEmpty"] = [this](FunctionCall const& _call) -> std::optional<bytes>
-	{
-		soltestAssert(_call.arguments.parameters.empty(), "No arguments expected.");
-		return toBigEndian(u256(storageEmpty(m_contractAddress) ? 1 : 0));
+	return {
+		{
+			"smokeTest",
+			[](FunctionCall const&) -> std::optional<bytes>
```
Comment (cameel) [inline test/libsolidity/SemanticTest.cpp]:
```suggestion
			[](FunctionCall const&) -> optional<bytes>
```

Same below.

File: test/libsolidity/SemanticTest.h [hunk 2]
```diff
@@ -80,10 +80,11 @@ class SemanticTest: public SolidityExecutionFramework, public EVMVersionRestrict
 private:
 	TestResult runTest(std::ostream& _stream, std::string const& _linePrefix, bool _formatted, bool _isYulRun, bool _isEwasmRun);
 	bool checkGasCostExpectation(TestFunctionCall& io_test, bool _compileViaYul) const;
-	void initializeBuiltins();
+	std::map<std::string, Builtin> makeBuiltins();
```
Comment (cameel) [inline test/libsolidity/SemanticTest.h]:
```suggestion
	std::map<std::string, Builtin> makeBuiltins() const;
```

**Review by cameel — DISMISSED** [review]
I have two small suggestions but other than that it's a straightforward refactor so marking as accepted already.

File: test/libsolidity/SemanticTest.h (hunk 2 above)
Comment (aarlt) [reply test/libsolidity/SemanticTest.h]: Some builtins are currently calling non-const functions, e.g. `balance` -> `ExecutionFramework::balanceAt(address)` and `storageEmpty` -> `ExecutionFramework::storageEmpty(m_contractAddress)`. But it looks like that they can be changed to be `const` too. I will update them in this PR.

**Review by chriseth — APPROVED** [review]

**Pull request merged**